ellipse_detection.py

```diff
@@ -0,0 +1,98 @@
+import numpy as np
+import cv2 as cv
+import matplotlib.pyplot as plt
+
+
+def center_crop(img, new_h, new_w):
+    old_h, old_w = img.shape[:2]
+    off_w = (old_w - new_w) // 2
+    off_h = (old_h - new_h) // 2
+    return img[off_h:off_h+new_h, off_w:off_w+new_w, :]
+
+
+def find_edges(img):
+    if len(img.shape) > 2:
+        img = cv.cvtColor(img, cv.COLOR_BGR2GRAY)
+    t, _ = cv.threshold(img, 0, 255, cv.THRESH_OTSU)
+    std = img.std()
+    edges = cv.Canny(img, min(255, t + 2 * std), max(0, t - 2 * std))

```

Comment:
just curious, where did you get that formula from?

Reply:
You have to provide two threshold to the Canny algorithm, a "high" primary threshold and a secondary "low" threshold, this is just a rule of thumb to pick a two reasonable values. The two values are just the upper and lower bound of the range where I expect to find 90% of the pixel intensities assuming that the intensities follow a gaussian distribution.

Reply:
Nice, that is good to know :) Feels like i lack a lot of statistical knowledge from my home university, we had hardly any courses there